.changeset/odd-wolves-sit.md

```diff
@@ -0,0 +1,5 @@
+---
+"@firebase/firestore": patch
+---
+
+Add assertion IDs and optional context objects that will be included in production log statements for fail and hardAsserts
```

Comment:
Consider a reword like this: Add unique IDs and state information into fatal error messages instead of the generic "unexpected state" message.

Reply:
Updated. Thanks for the suggestion.